core/autoscaling.go

```diff
@@ -620,7 +619,26 @@ func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) []str
 		return []string{baseInstance.typeInfo.instanceType}
 	}
 	return strings.Split(allowed, ",")
+}
+
+func (a *autoScalingGroup) getDisallowedInstanceTypes(baseInstance *instance) []string {
+	var disallowed, disallowedInstanceTypesTag string
+
+	// Check option of allowed instance types
+	// If we have that option we don't need to calculate the compatible instance type.
+	if tagValue := a.getTagValue("disallowed-instance-types"); tagValue != nil {
+		disallowedInstanceTypesTag = strings.Replace(*tagValue, " ", ",", -1)
+	}
+	disallowedInstanceTypes := strings.Replace(a.region.conf.DisallowedInstanceTypes, " ", ",", -1)
+
+	// Command line config has a priority
+	if disallowedInstanceTypes != "" {
+		disallowed = disallowedInstanceTypes
+	} else {
+		disallowed = disallowedInstanceTypesTag
```

Comment:
It's weird as we usually have the other type of behavior: the general arguments are usually here to allow general configuration while the ASG tags allow to override those specifically for certain ASG.